### PR TITLE
test: disable terser/minify by default

### PR DESCRIPTION
**Pull request by pimlie**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

We have probably all wasted hours of our time waiting for terser to run on the test fixtures. Thats completely useless in most cases because we dont have any need for it.

This PR speeds up the building of the fixtures with ~40 seconds for me (from 100-110s to 65s)

On CircleCI, the nighty build from last night:
image

This pr:
image

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing. **YES AND MUCH QUICKER**

### CODE REVIEW

File: test/utils/nuxt.js [hunk 1]
```diff
@@ -33,5 +33,14 @@ export const loadFixture = async function (fixture, overrides) {
   config.dev = false
   config.test = true
 
+  // disable terser to speed-up fixture builds
+  if (config.build) {
+    if (!config.build.hasOwnProperty('terser')) {
```
Comment (pi0) [inline test/utils/nuxt.js]: isn't config.build.terser enough?

**Review by pi0 — COMMENTED** [review]
Great! BTW why we need terser for modern and async-config?

**Comment by pimlie** [comment]
- modern needs it because it checks the bundle if it contains modern syntax. If we dont use terser on that then you have to match all kind of whitespace stuff.

This is part of the output from one of the failing tests when I didnt had terser enabled:
```
    /* harmony default export */ var pagesvue_type_script_lang_js_ = ({
      test() {
        var test = {
          build: true,
          test: true,
          arrow: () => {
            return 'build test';
          }
        };
        return {
          test
        };
      }·
    });

```

while the test searches if the output contains `arrow:()=>\"build test\"`. We could change it to just check the `arrow: () => {` line, should be ok I guess?

- async-config needs it due to the `async-config.size-limit` test. Not sure if that could be changed or not

**Comment by pi0** [comment]
Something more general: Why not automatically optimizing defaults when env is test (config.test) This can make module tests faster.

File: test/utils/nuxt.js (hunk 1 above)
Comment (pimlie) [reply test/utils/nuxt.js]: You mean instead of `hasOwnProperty`? Probably yes, will change that

**Comment by pi0** [comment]
> async-config needs it due to the async-config.size-limit test. Not sure if that could be changed or not

I'm not sure if that test is even relevant to this test. Please add a comment above it or skip for now. After having perf reports we only build a small fixture with terser to benchmark.

**Comment by pi0** [comment]
For modern, if we update tests that would be great. Saving even more time :)

**Comment by pimlie** [comment]
Yes, we broke the 200s mark on Circle CI!

image

**Comment by pimlie** [comment]
> Something more general: Why not automatically optimizing defaults when env is test (config.test) This can make module tests faster.

I find it sometimes confusing when Nuxt silently changes default settings based on the environment.

Think a setting like this would be much more suited to be changed in one of the test-utils packages.

**Review by pi0 — APPROVED** [review]

**Pull request merged**